modules/drivers/sparksql/src/metabase/driver/hive_like.clj

```diff
@@ -158,6 +160,82 @@
     (recur driver hsql-form (* amount 3) :month)
     (hx/+ (hx/->timestamp hsql-form) (hsql/raw (format "(INTERVAL '%d' %s)" (int amount) (name unit))))))
 
+(defmethod sql.qp/->honeysql [:sparksql :datetime-diff]
+  [driver [_ x y unit]]
+  (let [x (sql.qp/->honeysql driver x)
+        y (sql.qp/->honeysql driver y)]
+    (case unit
```

Comment:
Is this missing the error handling, or is it not needed here?

Reply:
The error handling was intended to catch time type args, which sparksql doesn't have. But it will go through the default error handling now anyway, which can catch other types of args now.